packages/@ember/application/instance.ts

```diff
@@ -120,15 +124,23 @@ const ApplicationInstance = EngineInstance.extend({
     this._booted = true;
 
     return this;
-  },
+  }
 
-  setupRegistry(options) {
-    this.constructor.setupRegistry(this.__registry__, options);
-  },
+  setupRegistry(options: BootOptions) {
+    (this.constructor as typeof ApplicationInstance).setupRegistry(this.__registry__, options);
+  }
+
+  _router?: Router;
```

Comment:
Given how this is used, it *shouldn't* matter over-much, but it's worth note that this results in a `defineProperty` emit when targeting older ES versions, and the corresponding semantics on later ES versions.